internal/controller/suite_test.go

```diff
@@ -67,19 +71,28 @@ import (
 // http://onsi.github.io/ginkgo/ to learn more about Ginkgo.
 
 var (
-	cfg            *rest.Config
-	k8sClient      client.Client
-	testEnv        *envtest.Environment
-	gitServer      *http.Server
-	gitStoragePath string
-	cancel         context.CancelFunc
-	ctx            context.Context
-	gitServerPort  string
+	cfg              *rest.Config
+	cfgDev           *rest.Config
+	cfgStaging       *rest.Config
+	k8sClient        client.Client
+	k8sClientDev     client.Client
+	k8sClientStaging client.Client
+	testEnv          *envtest.Environment
+	testEnvDev       *envtest.Environment
+	testEnvStaging   *envtest.Environment
+	gitServer        *http.Server
+	gitStoragePath   string
+	cancel           context.CancelFunc
+	ctx              context.Context
+	gitServerPort    string
 )
 
 const (
-	EventuallyTimeout   = 90 * time.Second
-	WebhookReceiverPort = 3333
+	EventuallyTimeout         = 90 * time.Second
+	WebhookReceiverPort       = 3333
+	kubeconfigSecretNamespace = "default"
+	kubeconfigSecretLabel     = "kubeconfig"
```

Comment:
lets move these to the constants packages, create a new file called configurations.go